### ISSUE TITLE
Transformer-base standard configuration?

### ISSUE THREAD

**Issue by emjotde** [awslabs/sockeye#762]
Hi,
I am trying to set up a couple of reference training configurations across different toolkits -- as close to the original transformer-base configuration as possible. This is what I have so far for Sockeye. Is there anything in there that would ring your alarm bells? Thanks a lot!

Validation based on BLEU is intentional. 

```
sockeye-train --device-ids $GPUS \                                                                                                                     
    --source $DATA/train.spm.en --target $DATA/train.spm.de --max-seq-len 200:200 \                                                                    
    --shared-vocab --num-words 32000:32000 \                                                                                                           
    --validation-source $DATA/valid.spm.en --validation-target $DATA/valid.spm.de \                                                                    
    --output model \                                                                                                                                   
    --encoder transformer --decoder transformer --num-layers 6:6 --num-embed 512:512 \                                                                 
    --transformer-model-size 512 --transformer-attention-heads 8 --transformer-feed-forward-num-hidden 2048 \                                          
    --transformer-preprocess "" --transformer-postprocess drn \                                                                                        
    --transformer-dropout-attention 0 --transformer-dropout-act 0 --transformer-dropout-prepost 0.1 --label-smoothing 0.1 \                            
    --weight-tying --weight-tying-type src_trg_softmax \                                                                                               
    --weight-init xavier --weight-init-scale 3.0 --weight-init-xavier-factor-type avg \                                                                
    --gradient-clipping-threshold -1 \                                                                                                                 
    --initial-learning-rate 0.0003 --batch-type word --batch-size 8192 \                                                                               
    --learning-rate-scheduler-type fixed-rate-inv-sqrt-t --learning-rate-warmup 8000 \                                                                 
    --optimizer adam --optimized-metric bleu \                                                                                                         
    --checkpoint-interval 5000 --max-num-checkpoint-not-improved 20 --checkpoint-frequency 5000 --decode-and-evaluate 500 --keep-last-params 60 \      
    2>&1 | tee train.log     
```

**Comment by fhieber** [comment]
Hi,

yes, that would most likely be configuration closest to Vaswani et al 2017's base transformer. I am not sure whether they set early stopping criteria, you might want to use something like `--max-updates` instead of `--max-num-checkpoint-not-improved 20`? 

Also, Sockeye writes a log file by default in `model/log`, you can leave out the pipe to `tee` :)

**Comment by fhieber** [comment]
However, default arguments for Sockeye transformers (i.e. positions of layer normalizations in layers) are intentionally set as is as they tend to perform better than the original Vaswani base transformer. But I am sure you are aware of this.

**Comment by emjotde** [comment]
Thanks a lot! Yes I am aware, although for more exotic models the orginal formulation has been consistently better for me. 

I am seeing this with warmup:
```
[INFO:sockeye.lr_scheduler] Learning rate warmup: 3140%
```
Anything to be worried about?

**Comment by fhieber** [comment]
That looks suspicious. We haven't given learning rate warmup a lot of love in the past as it tends to be not very useful for Sockeye. Let me take a look.

**Comment by emjotde** [comment]
Yes, this is after 260K iterations. The training looks fine. Is there a way to display the effective learning rate?

**Comment by fhieber** [comment]
It should be stored in the `metrics` file for every checkpoint

**Comment by emjotde** [comment]
Got it, looks good.

**Comment by fhieber** [comment]
Closing for now, feel free to reopen if you have more questions.
